packages/compose/src/hooks/use-focus-outside/index.ts

```diff
@@ -0,0 +1,173 @@
+/**
+ * External dependencies
+ */
+import type {
+	FocusEventHandler,
+	EventHandler,
+	MouseEventHandler,
+	TouchEventHandler,
+	FocusEvent,
+	MouseEvent,
+	TouchEvent,
+} from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useCallback, useEffect, useRef } from '@wordpress/element';
+
+/**
+ * Input types which are classified as button types, for use in considering
+ * whether element is a (focus-normalized) button.
+ */
+const INPUT_BUTTON_TYPES = [ 'button', 'submit' ];
+
+type FocusNormalizedButton =
```

Comment:
Could this also use inline docs?

Reply:
Done in https://github.com/WordPress/gutenberg/pull/45317/commits/9f5884e22ff85e8c5f99ca06870c962482694339